UnitTests/ScientistTests.cs

```diff
@@ -229,5 +229,24 @@ public void RunsBothBranchesOfTheExperimentWithIEqualitySetAndReportsFailure()
             mock.Received().Candidate();
             Assert.False(TestHelper.Results<ComplexResult>().First().Matched);
         }
+
+        [Fact]
+        public void RunsBeforeRun()
+        {
+            // We introduce side effects for testing. Don't do this in real life please.
+            var mock = Substitute.For<IControlCandidate<int>>();
+            mock.Control().Returns(42);
+            mock.Candidate().Returns(42);
+            
+            var result = Scientist.Science<int>("beforeRun", experiment =>
+            {
+                experiment.BeforeRun(mock.BeforeRun);
+                experiment.Use(mock.Control);
+                experiment.Try(mock.Candidate);
+            });
+
+            Assert.Equal(42, result);
+            mock.Received().BeforeRun();
```

Comment:
This doesn't test that it actually happened _before_ the run. Just that it happened. Tests that ensure ordering tend to be tricky and fragile so I'll merge this and we can consider if such a test is worth it later.